lib/aur-chroot

```diff
@@ -260,12 +267,32 @@ if (( print_path )); then
 fi
 
 if (( build )); then
-    # use makechrootpkg -c as default build command (sync /root container)
-    # arguments after -- are used as makechrootpkg arguments
-    # shellcheck disable=SC2086
-    $AUR_PACMAN_AUTH "${auth_args[@]}" makechrootpkg -r "$directory" \
-         "${bindmounts_ro[@]/#/-D}" "${bindmounts_rw[@]/#/-d}" \
-         "${makechrootpkg_args[@]}" -- "${makechrootpkg_makepkg_args[@]}"
+    bindmounts_rw+=(
+        "$PWD"
+        "${PKGDEST:-$PWD}:/pkgdest"
+        "${LOGDEST:-$PWD}:/logdest"
+        "${SRCDEST:-$PWD}:/srcdest"
+        "${SRCPKGDEST:-$PWD}:/srcpkgdest"
+    )
+
+    printf "u %q $(id -u) \"makepkg user\" /build /bin/bash\n" "$makepkg_user" | \
+        $AUR_PACMAN_AUTH systemd-sysusers --root "$directory"/root --replace=/usr/lib/sysusers.d/build.conf -
+
+    $AUR_PACMAN_AUTH systemd-nspawn \
+        -D "$directory"/root \
+        --chdir="$PWD" \
+        --user="$makepkg_user" \
+        --volatile=overlay \
+        --setenv=PKGDEST=/pkgdest \
```

Comment:
These variables should match bindmounts_rw, eg. `--setenv=PKGDEST=${PKGDEST:-$PWD}:/pkgdest"`. The variables could be assigned in the script to avoid repetition.

Reply:
Does makepkg support colon separated paths in environment variables? How does it work? Does it use the first one that exists? Nice to know btw (couldn't find it in the documentation)

The reason why I implemented it like this is that:
- I need to bind mount all of PKGDEST LOGDEST SRCDEST SRCPKGDEST
- I don't know which ones are going to be pointing at the same directory (by default $PWD is used for all 4)
- So I mount all the paths (which might be the same path 4 times) in 4 different fixed locations and pass those to makepkg

Sorry for the mansplaining :) But given that, why do I have to pass the path outside the chroot to the makepkg variables too?

Reply:
That was a mistake on my behalf: it should be `--setenv=PKGDEST=${PKGDEST:-$PWD}`. In any case, `aur-build` sets PKGDEST to a temporary directory, so that needs to be propagated to `makepkg` (if possible, to avoid manually moving packages like `devtools` does.)

Reply:
Sorry, I feel like an idiot: I'm setting environment variables to directories with fixed names because e.g. `/pkgdest` is the path *inside* the container (i.e. the mount target), the mappings are added to the `bindmounts_rw` variable just above the `systemd-nspawn` call.
Then, the path *inside* the container will always be /pkgdest, but outside the container the location will be the correct one (i.e. PWD or the PKGDEST variable)
I have actually tested the integration with `aur build` and it looks like its working as intended, please enlighten me :D